src/App.tsx

```diff
@@ -1,204 +1,118 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import React, { useCallback, useEffect, useState } from 'react';
+import { getTodos, USER_ID } from './api/todos';
+
+import { Error } from './components/Errors';
+import { TodoFooter } from './components/TodoFooter';
+import { TodoHeader } from './components/TodoHeader';
+import { TodoList } from './components/TodoList';
+import { UserWarning } from './components/UserWarning';
+
+import { FILTER_TYPES } from './types/FilterType';
+
+import { Todo } from './types/Todo';
+import { ErrorMessage } from './types/types';
 
 export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  // const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+
+  const [isLoading, setIsloading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    setIsloading(true);
+    getTodos()
+      .then(setTodoList)
+      .catch(() => setErrorMessage(ErrorMessage.Get))
+      .finally(() => setIsloading(false));
+  }, []);
+
+  const handleHideError = () => {
+    setErrorMessage('');
+  };
+
+  const handleError = (error: string) => {
+    setErrorMessage(error);
+  };
+
+  const handleAddNewTodo = (newTodo: Todo) => {
+    setTodoList(prevTodos => [...prevTodos, newTodo]);
+  };
+
+  const markAllTodoCompleted = () => {
+    const isCompleted = todoList.every(todo => todo.completed);
+
+    const updatedTodo = todoList.map(todo => ({
+      ...todo,
+      completed: !isCompleted,
+    }));
+
+    setTodoList(updatedTodo);
+  };
+
+  const handleChangeToggle = (id: number) => {
+    setTodoList(prevTodo =>
+      prevTodo.map(todo =>
+        todo.id === id ? { ...todo, completed: !todo.completed } : todo,
+      ),
+    );
+  };
+
+  const handleDeleteTodo = (id: number) => {
+    setTodoList(prevTodo => prevTodo.filter(todo => todo.id !== id));
+  };
+
+  const handleDeleteCompletedTodos = () => {
+    setTodoList(todoList.filter(todo => !todo.completed));
+  };
+
+  const handleFilterTodo = useCallback(
+    (option: string) => {
+      switch (option) {
+        case FILTER_TYPES.ALL:
+          return todoList;
+        case FILTER_TYPES.ACTIVE:
+          return todoList.filter(todo => !todo.completed);
+        case FILTER_TYPES.COMPLETED:
+          return todoList.filter(todo => todo.completed);
```

Comment:
filtering not works.

lift option state up  to this component and filter todos based on option state.
in TodoList component you should pass filteredTodos.

also check styles for active filter link

https://github.com/user-attachments/assets/f9daa0ad-6a73-41a4-b03b-60b0f5a70845